pkg/gvk/gvk.go

```diff
@@ -19,12 +20,39 @@ var crdGVR = schema.GroupVersionResource{
 	Resource: "customresourcedefinitions",
 }
 
+type cohabitatingResource struct {
+	resource       string
+	groupResource1 schema.GroupResource
+	groupResource2 schema.GroupResource
+	seen           bool
+}
+
+func newCohabitatingResource(resource, group1, group2 string) *cohabitatingResource {
+	return &cohabitatingResource{
+		resource:       resource,
+		groupResource1: schema.GroupResource{Group: group1, Resource: resource},
+		groupResource2: schema.GroupResource{Group: group2, Resource: resource},
+		seen:           false,
+	}
+}
+
+func NewCohabitatingResources() map[string]*cohabitatingResource {
+	return map[string]*cohabitatingResource{
```

Comment:
FYI for anyone looking on, this is borrowed from Velero source.

Our goal here is to accurately simulate the logic that Velero run to choose which resource to back up when cohabitating API groups are present on a source cluster.